### PR TITLE
Nickakhmetov/HMP-194 Configure Linters / HMP-96 Introduce Typescript

### PR DESCRIPTION
**Pull request by NickAkhmetov**
This PR includes various linting configuration changes, which I've tried to document as fully as possible [here](https://hms-dbmi.atlassian.net/wiki/spaces/GL/pages/3009642572/Tech+Updates+Progress+Tracking#Linting-Revisions).

I also started adding TypeScript to our project; my full motivation for this is documented in the linked doc, but essentially this was a necessary step to take in order to get rid of babel.

This cannot be merged until the resulting lint errors are resolved, which I'll take care of in another upcoming PR.

6,761 of the total changed lines in this PR are in the `package-lock.json`; the rest adds up to ~200 additions/deletions combined.

### CODE REVIEW

File: context/.swcrc [hunk 1]
```diff
@@ -11,6 +11,7 @@
             "decorators": false,
             "decoratorsBeforeExport": false
         },
+        "externalHelpers": true,
```
Comment (john-conroy) [inline context/.swcrc:14]: Do we need to install `@swc/helpers` as https://swc.rs/docs/configuration/compilation#jscexternalhelpers indicates?

**Review by john-conroy — COMMENTED** [review]
It predates this work, but we moved e2e testing/artillery out of `context` and into the `end-to-end` dir due to some complications with our CI. Is there configuration to make these eslint changes apply to those directories?

https://eslint.org/docs/latest/use/configure/configuration-files#cascading-and-hierarchy.

File: context/.swcrc (hunk 1 above)
Comment (NickAkhmetov) [reply context/.swcrc:14]: Yes, this was installed in the other branch once I realized it was missing: https://github.com/hubmapconsortium/portal-ui/blob/02a4743766c2035a503482f327b5cfd47fe2c0f0/context/package.json#L89

This ended up being necessary to get tests to run successfully.

File: CHANGELOG-lint-updates.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- Updated project ESLint configuration and dependencies
+- Removed Babel
+- Introduced TypeScript
```
[the remark this answers is not in the document]
Comment (NickAkhmetov) [reply CHANGELOG-lint-updates.md]: I think that would be helpful. I'll look into extensions to help with enforcing those conventions as well.

**Review by john-conroy — DISMISSED** [review]

**Review by john-conroy — APPROVED** [review]

**Pull request merged**